pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1046,13 +1046,16 @@ self: super: {
   # test suite requires git and does a bunch of git operations
   restless-git = dontCheck super.restless-git;
 
-  # requires git at test-time *and* runtime, but we'll just rely on users to
-  # bring their own git at runtime. Additionally, sensei passes `-package
-  # hspec-meta` to GHC in the tests, but doesn't depend on it itself.
-  sensei = overrideCabal (drv: {
+  # patch out a flaky test that depends on output from hspec >= v2.11.7.
+  # https://github.com/hspec/sensei/issues/125
+  sensei = appendPatch ./patches/sensei.patch (overrideCabal (drv: {
```

Comment:
We generally try to pull patches from upstream if at all possible.  There should be a few examples of `fetchpatch` in this file showing creative ways of pulling from individual commits or PRs (even targeting individual files within a commit).

Would it be possible to change this to a use of `fetchpatch`?

Reply:
This is a new issue, so a commit that fixes the failing build does not exist in upstream yet. I opened the issue there.
hspec/sensei#125

Reply:
Maybe you could send a PR to `sensei` showing sol what you're expecting the change to be?  And then you can pull in the diff from that PR with `fetchpatch`?

Alternatively, maybe you could just set `sensei` as `dontCheck`?  You could argue that the bounds on `hspec` are incorrect in `sensei.cabal` (which we often work around here in Nixpkgs just by setting `dontCheck` when it affects tests).